rest_spec.md

```diff
@@ -63,15 +63,15 @@ These APIs allow the admin user to create new users with access to certain resou
 
 | HTTP Verb | URL | Request Body | Response Code | Response Body | Function |
 | --- | --- | :---  | :---  | :---  | :--- |
-| POST | `/users` | `{"roleSetId": "client\|admin","resourceId": "*\|some id"}` | See Collection responses | `{“bearerToken”: "token"}` | Create a new user and assign a bearer token, setup internal management of the user-id that does not require saving the bearer token and attached the named `roleSet` for the `resource-id` to the new user |
-| DELETE | `/users/user-id` | `{"roleSetId": "client\|admin", "resourceId": "*\|<some-user-id>"}` | See Item responses | `{"bearerToken": "token"}` | Removes the user and all access rules from the system |
+| POST | `/users/user-id/permissions` | `{"userId": "user-id", "roleSetId": "client\|admin","resourceId": "*\|some id"}` | See Collection responses | `{"userId": "user_id", “bearerToken”: "token"}` | Create a new user and assign a bearer token and user-id, setup internal management of the user-id that does not require saving the bearer token and attached the named `roleSet` for the `resource-id` to the new user |
+| DELETE | `/users/user-id/permissions` | `{"roleSetId": "client\|admin", "resourceId": "*\|<some-user-id>"}` | See Item responses | `{"bearerToken": "token"}` | Removes the user and all access rules from the system |
 
 # Auth-Server REST API (Private)
 
 The Auth-Server is a microservice that Harness uses to manages `User` resources and the routes and resoures they are authorized to access. It is secured with connection security no TLS or Auth itself is required and no client is provided since the users need neve access it direclty. **This section is under construction.**
 
 | HTTP Verb | URL | Request Body | Response Code | Response Body | Function |
 | --- | --- | :---  | :---  | :---  | :--- |
-| POST | `/authenticate` | `{"token": "string"}` | 200 or 401 | `{"accessToken": "string", "ttl": <optional long>, "refreshToken": "optional string"}` | authenticates user's access and returns a session token |
-| POST | `/authorize` | `{"accessToken": "string"`, ` "role": "string"`, ` "resourceId": "string"}` | 200 or 403 | `{"success": "true"}` | Given a session/access token authorize the access requested or return an error code |
+| POST | `/auth/token` | `grant_type=password&username=user-id&password=granted-token`, also app server's credentials should be provied by Authorization Basic header (see [https://tools.ietf.org/html/rfc6749#section-4.3] for details) | 200 or 401 | `{"access_token": "string", "token_type": "", "refresh_oken": "optional string"}` | authenticates user's access and returns a session token |
```

Comment:
shouldn't this be `POST /token` since we removed the `/auth` from URIs

I've incorporated these, will update develop with my next push